src/common/type/pack.c

```diff
@@ -0,0 +1,1513 @@
+/***********************************************************************************************************************************
+Pack Type
+
+Each pack field begins with a one byte tag. The four high order bits of the tag contain the field type (PackType). The four lower
+order bits vary by type.
+
+Integer types (packTypeData[type].valueMultiBit) when the value is >= -1 and <= 1:
+  3 more value indicator set to 0
+  2 value low-order bit
+  1 more ID delta indicator
+  0 ID delta low order bit
+
+Integer types (packTypeData[type].valueMultiBit) when the value is < -1 or > 1:
+  3 more value indicator set to 1
+  2 more ID delta indicator
+0-1 ID delta low order bit
+
+String and binary types (packTypeData[type].valueSingleBit):
+  3 value bit
+  2 more ID delta indicator
+0-1 ID delta low order bits
+
+Array and object types:
+  4 more ID delta indicator
+0-3 ID delta low order bit
```

Comment:
I thought we were referring to 4 bits, not 5, so why are we referencing "4" and then "0-3"?

Reply:
Fixed in f50252aa78e9ef4.